Formula/s/senpai.rb

```diff
@@ -0,0 +1,23 @@
+class Senpai < Formula
+  desc "Modern terminal IRC client"
+  homepage "https://sr.ht/~taiite/senpai/"
+  url "https://git.sr.ht/~taiite/senpai/archive/v0.2.0.tar.gz"
+  sha256 "9786fd83f3e1067549c3c88455a1f66ec66d993fe597cee334d217a5d1cf4803"
+  license "ISC"
+
+  depends_on "go" => :build
+  depends_on "scdoc" => :build
+
+  def install
+    system "make"
+    system "make", "PREFIX=#{prefix}", "install"
+  end
+
+  def uninstall
+    system "make", "PREFIX=#{prefix}", "uninstall"
+  end
+
+  test do
+    assert_equal "#{HOMEBREW_PREFIX}/bin/senpai", shell_output("which senpai").strip
```

Comment:
We need a test that exercises the some of the functionality of the app. Version checks or usage checks (foo --version or foo --help) are not sufficient, as explained in the formula cookbook.

In most cases, a good test would involve running a simple test case: run #{bin}/foo input.txt.

- Then you can check that the output is as expected (with assert_equal or assert_match on the output of shell_output)
- You can also check that an output file was created, if that is expected: assert_predicate testpath/"output.txt", :exist?

Some advice for specific cases:

- If the formula is a library, compile and run some simple code that links against it. It could be taken from upstream's documentation / source examples.
- If the formula is for a GUI program, try to find some function that runs as command-line only, like a format conversion, reading or displaying a config file, etc.
- If the software cannot function without credentials, a test could be to try to connect with invalid credentials (or without credentials) and confirm that it fails as expected.
- Same if the software requires a virtual machine, docker instance, etc. to be running.

Reply:
This could be difficult. The app doesn't actually expose any command line options. Running the command simply launches the window, and there's not even a `version` flag to check hence the install location check here.

Reply:
there are some GUI test in the repo that you can take reference

Reply:
I've opened up a discussion with the devs about cutting a new release for this. At the moment, the program fails to generate its config file so the only thing to test against it an error message. I know that this is fixed in the main branch, so I'll look at reopening this when we have something more up-to-date.